src/main/java/frc/robot/RobotContainer.java

```diff
@@ -108,15 +110,21 @@ private void configureBindings() {
         // Uncomment this to calibrate the wrist angle
         // shooterSubsystem.setDefaultCommand(new CalibrateWristAngleCommand(shooterSubsystem));
 
+        intakeSubsystem.setDefaultCommand(commands.waitForIntakeCommand());
+
+        driverController.a().toggleOnTrue(ejectCommand);  // Allow ejecting a note to be stopped on a second a press
+        driverController.b().toggleOnTrue(commands.intakeNoteAndKeepRunningCommand());
+        driverController.x().and(shooterSubsystem::isNoteInShooter).onTrue(trapShoot);
+        driverController.y().and(shooterSubsystem::isNoteInShooter).onTrue(stageSpeakerShoot);
 
-        driverController.leftBumper().onTrue(ampShoot);
-        driverController.rightBumper().onTrue(speakerShoot);
-        driverController.x().onTrue(trapShoot);
-        driverController.y().onTrue(stageSpeakerShoot);
-        driverController.a().onTrue(intakeSubsystem.eject(2)).onFalse(intakeSubsystem.eject(0));
-        driverController.b().onTrue(longShot);
-        driverController.rightTrigger().onTrue(shootCommand);
-        driverController.leftTrigger().onTrue(stopShoot);
+        driverController.leftBumper().and(shooterSubsystem::isNoteInShooter).onTrue(ampShoot);
+        driverController.rightBumper().and(shooterSubsystem::isNoteInShooter).onTrue(speakerShoot);
+        driverController.leftTrigger().and(shooterSubsystem::isNoteInShooter).onTrue(stopShoot);
```

Comment:
Do we need to have the note in the shooter to stop the shoot? That seems like possibly a bad idea.

Reply:
Yeah I got caught here @Jnesselr as well

Reply:
Nope that's a good call. Probably not.